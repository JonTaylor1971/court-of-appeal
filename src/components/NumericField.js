import React, { Component } from 'react';

class NumericField extends Component {

    constructor(props) {
        super(props);
        this.fieldChanged = this.fieldChanged.bind(this);
    }

    render() {
        let autofocus = this.props.autofocus ? true : false;

       return <input
           id={this.props.id}
           name={this.props.name}
           autoFocus={autofocus}
           ref={this.props.ref}
           value={this.props.value || ''}
           onChange={this.fieldChanged}
           placeholder={this.props.placeholder}
           onKeyPress={this.props.handleKeyPress}
       />;
    }

    /**
     * Remove all non-numeric characters
     * @param e - the change event
     */
    fieldChanged(e) {
        e.target.value = e.target.value.replace(/\D/g,'');
        this.props.fieldChanged(e);
    }
}
export default NumericField;
