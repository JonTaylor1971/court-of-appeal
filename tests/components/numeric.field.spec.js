import React from 'react';
import NumericField from '../../src/components/NumericField';
import renderer from 'react-test-renderer';

test('should only allow numeric entries', ()=> {

    let value = '';
    let fieldChanged = (e) => { value = e.target.value; };

    const numericField = renderer.create(
        <NumericField id="number-field" value={value}
        fieldChanged={fieldChanged.bind(this)}/>,
    );
    let tree = numericField.toJSON();
    expect(tree).toMatchSnapshot();

    let e = { target: { value: "42"}};
    tree.props.onChange(e);
    expect(value).toMatch('42');

    e = { target: { value: "bah" } };
    tree.props.onChange(e);
    expect(value).toMatch('');

    e = { target: { value: "CA1234" } };
    tree.props.onChange(e);
    expect(value).toMatch('1234');

    e = { target: { value: "*bah1#?234?wtf?" } };
    tree.props.onChange(e);
    expect(value).toMatch('1234');

});
