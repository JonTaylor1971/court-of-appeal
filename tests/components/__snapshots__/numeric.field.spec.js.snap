// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should only allow numeric entries 1`] = `
<input
  autoFocus={false}
  id="number-field"
  name={undefined}
  onChange={[Function]}
  onKeyPress={undefined}
  placeholder={undefined}
  value=""
/>
`;
